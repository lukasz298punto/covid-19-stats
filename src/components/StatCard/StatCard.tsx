import { Card, CardProps } from 'antd';
import styled from 'styled-components';

export type Props = {
    kind?: 'success' | 'warning' | 'error';
    value: number;
} & CardProps;

const getRoleColor = (kind: Props['kind']) => {
    if (!kind) return null;

    const color = window
        .getComputedStyle(document.documentElement)
        .getPropertyValue(`--${kind}-color`);

    return color;
};

function StatCard({ value = 0, className, ...props }: Props) {
    return (
        <Card {...props} className={className}>
            {value}
        </Card>
    );
}

export default styled(StatCard)`
    border-color: ${({ kind }) => getRoleColor(kind)};
`;
