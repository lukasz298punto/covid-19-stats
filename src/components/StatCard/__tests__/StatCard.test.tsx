import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { StatCard } from 'components/StatCard';
import 'jest-styled-components';

it('renders app', () => {
    render(<StatCard value={1} />);
});

it('could accept none react element icon', () => {
    const component = renderer.create(<StatCard value={1} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
