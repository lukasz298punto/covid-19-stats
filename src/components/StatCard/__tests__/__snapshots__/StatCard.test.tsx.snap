// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`could accept none react element icon 1`] = `
.c0 {
  background-color: red;
}

<div
  className="ant-card ant-card-bordered c0 mt-5"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    1
  </div>
</div>
`;
